lib/autoload/trackable_job.rb

```diff
@@ -0,0 +1,69 @@
+# This is a mix-in that allows jobs to be trackable. Trackable jobs can be queried at /jobs/<id>.
+# A client requesting HTML would see a progress bar; a client requesting JSON will get a status
+# message. When the job is complete, the trackable job can specify a path to redirect the user to.
+#
+# To use this mix-in, implement +perform_tracked+ instead of +perform+, with the same arguments.
+# If you intend to use +rescue_from+ and you wish for the job to fail, call +rescue_tracked+ with
+# the exception.
+#
+# Jobs are ephemeral; do not expect jobs to remain after a long period of time. Also since jobs
+# are uniquely identified only by ID and do not need authentication, do not leak anything
+# sensitive from the job.
+module TrackableJob
+  extend ActiveSupport::Concern
+
+  class Job < ActiveRecord::Base
+    enum status: [:submitted, :completed, :errored]
```

Comment:
OK, so as discussed, we won't have `:processing` state and will not report progress to user. 

This also works for file downloading: If the user didn't see progress and click the button again, we won't re-download the files since they are cached.